Schema plugin for iso19139.ca.HNAP (initial version)

davidblasby opened the pull request:
This is the initial version of the iso19139.ca.HNAP schema.

This work was done by Canada's Federal Geospatial Portal (FGP) with additions from multiple people.
![image](https://user-images.githubusercontent.com/48937730/74340720-baac8600-4da6-11ea-9740-42783fa1872d.png)

josegar74 committed:
Rename of the profile, added also langId_from_gmdlanguage19139 to get the schema working. To remove when integrated https://github.com/geonetwork/core-geonetwork/pull/4207

josegar74 committed:
Update fixed info process - load English codelists as in HNAP contain English and French texts and expected UI language value /root/env/lang is not provided without further Java changes

Commit:
Removed $schemadir dependency

Commit:
Remove target as it can be built.

Commit:
remove intellij files and add to .ignore list.

Commit:
Remove xml spy file and intellij file.

Commit:
Use mainlanguage to determine language instead of gmd:LocalisedCharacterString

Commit:
gmd:LocalisedCharacterString should be based on the $langid so that it does not fail when there are multiple gmd:LocalisedCharacterString

Commit:
nap should be ca.HNAP even if it is a comment

Commit:
Remove debug message

Commit:
Change namespace XslUtil to XslUtilHnap and java to XslUtil to avoid confusion with other XslUtil

Commit:
Update thesaurus so that it uses mainlang/altlang
Also removed logic related to $lang as it should not be different based on your interface language

Commit:
Removed dependency on getCodeListFileUri and use getCodelistTranslation instead.
Also removed lang as no updates should be performed based on the interface language!
Also streamlined the IC code selection selection so that the code is easier to read.
Update - gmd:MD_CharacterSetCode for gmd:PT_Locale and gmd:MD_Metadata was being set to IC_95 so it should parsed as well to ensure it remains.
Added missing gmd:Country - IC_117 and gmd:fileType - IC_115

Commit:
Fixed bug with gmd:LocalisedCharacterString that should skip the main language or it could cause errors.
Also removed xsl:template  match="gmd:LocalisedCharacterString" as it was causing the languages to be processed twice adding multiple languages.

Commit:
Opps - forgot to create mainLanguage variable in last commit.

Commit:
Reverted some changes that were removed from the core geonetwork

Commit:
Could not use getCodelistTranslation as expected since this got the <label> and we need the <value>
So reverted some changes to get the codelist file manually.
Fixed issues with the gmd:Country code not being set correctly because there was not codelist table.
Fixed bug with gmd:fileType which should use the napm:napMD_FileFormatCode code table and add a gco:CharacterString

Commit:
Update some editor configuration so tabs are 2 char for xml files.

Commit:
DFO code lists are not used in this profile at this time

Commit:
Add license and readme file from the metadata 101 web site.

Commit:
Fixed some bugs related to selecting the gmd:LocalisedCharacterString without specifying local - causing errors as it would return english and french version.

Commit:
Fixed Thesaurus bugs - it was not loading the Thesaurus properly because the ThesaurusDir was empty
skos:concept should have been skos:Concept (Capital C) - it was not finding the record.
"Government of Canada Core Subject Thesaurus" was outside the <xsl:choose>

Commit:
Move the schematron-rules-ec-multilingual.sch out of depreciated and moved schematron-rules-ec.sch to depreciated because it had napec components.
We will probably remove the depreciated shortly....

david-blasby committed:
Merge branch 'master' of ../../ian_allen/dfo-build into 3.8.x

davidblasby committed:
Update README.md

jodygarnett commented on .editorconfig:
```diff
@@ -0,0 +1,18 @@
+[*]
```
Is this some from specific editor?

ianwallen replied:
I believe it may be been copied from geonetwork core.
The following may help.
https://github.com/geonetwork/core-geonetwork/pull/1517

jodygarnett replied:
Thanks @ianwallen some of my co-workers have explained the duplication is by design presently. Although it may be possible to improve the level of reuse @josegar74 ?

jodygarnett commented on .gitignore:
```diff
@@ -21,3 +21,9 @@
+target/
+*.iml
+.*/
+target/
```
target is twice

jodygarnett commented on pom.xml:
```diff
@@ -0,0 +1,80 @@
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>findbugs-maven-plugin</artifactId>
```
find bugs is no longer maintained

jodygarnett commented on src/main/java/ca/gc/schema/iso19139hnap/util/XslUtilHnap.java:
```diff
@@ -0,0 +1,265 @@
+/*
+ * ISO19139.hnap xsl utilities
+ * Ported from XslUtil.java and utils-fn.xsl in ecc geonetwork
```
This should be an open source license

jodygarnett commented on src/main/plugin/iso19139.ca.HNAP/formatter/groovy/iso19139/SummaryFactory.groovy:
```diff
@@ -0,0 +1,232 @@
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+package iso19139
```
I did not expect groovy code, is this generic code or is it specific to HNAP

jodygarnett commented:
Casual review here, it appears that there is a lot of duplication with files copyright 2016 FAO. I do not know if this is expected practice when creating a schema plugin, or if the original can be reused and overridden ...

ianwallen commented on src/main/plugin/iso19139.ca.HNAP/extract-ckan-dataset.xsl:
```diff
@@ -0,0 +1,374 @@
+        <presentation_form><xsl:value-of select="$presentationFormCkan" /></presentation_form>
+
+        <!-- TODO: Fixed value to the code for ECCC, check for FGP -->
+        <owner_org>49E2ADF4-AD7A-43EB-85C8-6433D37ED62C</owner_org>
```
The following should be removed, There should not be any ECCC specific id's

josegar74 committed:
Lucene indexing - update main language calculation and fixes for codelists in the metadata editor

josegar74 committed:
Remove duplicated entry in .gitignore

josegar74 committed:
Add header with open source license to Java file

juanluisrp committed:
Fix schema name from iso19139.nap to iso19139.ec.HNAP

Fix the schema name. Renamed some values that pointed to old
iso19139.nap schema name.

jodygarnett commented:
Thank you for the fix @juanluisrp
